src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -343,69 +327,44 @@ object NamespaceWithWorkflow {
+   * top level MemberAccess ASTs because the sub-ASTs don't make sense w/o the context of the parent. For example
+   * if we have "input: var=ns.ns1.my_task" it does not make sense to validate "ns1.my_task" by itself as it only
+   * makes sense to validate that "ns.ns1.my_task" as a whole is coherent
    *
-   * MemberAccess ASTs are of the form lhs.rhs
+   * Run for its side effect (i.e. Exception) but we were previously using a Try and immediately calling .get on it
+   * so its the same thing
    */
-  def getCallFromMemberAccessAst(ast: Ast, workflow: Workflow, wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter): Try[Call] = {
-    def callFromName(name: String): Try[Call] = {
-      workflow.calls.find(_.name == name) match {
-        case Some(c: Call) => Success(c)
-        case _ => Failure(new SyntaxError(wdlSyntaxErrorFormatter.undefinedMemberAccess(ast)))
-      }
-    }
-    val rhs = ast.getAttribute("rhs").sourceString
-
-    /**
-     * The right-hand side of a member-access AST should always be interpreted as a String
-     * Sometimes, the left-hand side is itself a MemberAccess AST, like in the expression
-     * for `call t1` below.  In that example, callFromName("ns.ns2.task_name") would be
-     * called.  In the `call t2` example, callFromName("alias") is called
-     *
-     * import "test.wdl" as ns
-     * workflow w {
-     *  call ns.ns2.task_name
-     *  call t1 {
-     *    input: x=ns.ns2.task_name.output
-     *  }
-     *
-     *  call ns.ns2.task_name as alias
-     *  call t2 {
-     *    input: y=alias.output
-     *  }
-     *}
+  def validateMemberAccessAst(memberAccessAst: Ast,
+                              workflow: Workflow,
+                              errorFormatter: WdlSyntaxErrorFormatter): Unit = {
+    val memberAccess = MemberAccess(memberAccessAst)
+    /*
+      This is a shortcut - it's only looking at the workflow's locally qualified name and not the fully qualified
+      name. This is ok for now because we do not support nested workflows and all call names must be unique within
+      a workflow, however if we support nested workflows this will no longer work properly.
      */
-    val lhs = callFromName(ast.getAttribute("lhs") match {
-      case a: Ast => WdlExpression.toString(a)
-      case terminal: Terminal => terminal.sourceString
-    })
-
-    lhs match {
-      case Success(c: Call) =>
-        c.task.outputs.find {_.name == rhs}.getOrElse {
-          throw new SyntaxError(wdlSyntaxErrorFormatter.memberAccessReferencesBadTaskInput(ast))
-        }
-        Success(c)
-      case f => f
+    val call = workflow.calls find { _.name == memberAccess.lhs }
+
+    call match {
+      case Some(c) if !c.task.outputs.exists(_.name == memberAccess.rhs) =>
+        throw new SyntaxError(errorFormatter.memberAccessReferencesBadTaskInput(memberAccessAst))
+      case None =>
+        throw new SyntaxError(errorFormatter.undefinedMemberAccess(memberAccessAst))
+      case _ => ()
```

Comment:
I think it would be more straightforward if the success case were handled first

```
case Some(c) if c.task.outputs.exists(_.name == memberAccess.rhs) => ()
...
```